src/system_info/system_info_sim_ivi.cc

```diff
@@ -23,11 +14,18 @@ SysInfoSim::SysInfoSim()
       msin_(""),
       spn_(""),
       conn_(NULL) {
-  InitDbusConnection();
+  conn_ = system_info::GetDbusConnection();
+  if (!conn_)
+    return;
+  modem_path_ = system_info::OfonoGetModemPath(conn_);
```

Comment:
No, the whole point is to check run-time. Modems can be powered up and down and enabled/disabled. Saving the dbus call means implementing state management here, which is more complex than a sparse call.

However, in the system_info::OfonoGetModemPath() method, it doesn't seem correct to return NULL (should return std::string value).